juju/juju
# add juju-updateseries command to assist restarting juju agents after series upgrade

## Description of change

Add a juju command available on each unit to assist with restarting juju agents after a series upgrade from trusty to xenial.  This requires writing systemd service files for the agents, link and enable them, and copying tools to a correctly named directory and relinking agent tools.  Optional is to restart the agents, if systemd is really running.

## QA steps

1. bootstrap juju localhost
2. juju deploy ubuntu -n 2 --series trusty
3. juju ssh 0 -- sudo do-release-upgrade -f DistUpgradeViewNonInteractive
4. juju ssh 1 -- sudo do-release-upgrade -f DistUpgradeViewNonInteractive
4. juju ssh 0 -- sudo rm /var/lib/juju/agents/unit-ubuntu-0/charm/wheelhouse/.bootstrapped
4. juju ssh 1 -- sudo rm /var/lib/juju/agents/unit-ubuntu-0/charm/wheelhouse/.bootstrapped
5. juju ssh 0 -- sudo shutdown -r now
6. juju ssh 0 -- sudo juju-updateseries --to-series xenial --from-series trusty
6. juju ssh 0 -- sudo juju-updateseries --to-series xenial --from-series trusty  --start-agents
6. juju ssh 1 -- sudo juju-updateseries --to-series xenial --from-series trusty  --start-agents  <-- won't start agents
7. juju ssh 1 -- sudo shutdown -r now

Once juju and the agents settle... juju status should show good results for the two units

## Documentation changes

The current instructions will need to be updated.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1749201
